packages/appium/lib/extension/extension-config.js

```diff
@@ -0,0 +1,458 @@
+        val: mainClass,
+      });
+    }
+
+    return problems;
+  }
+
+  /**
+   * @abstract
+   * @param {ExtData<ExtType>} extData
+   * @returns {Problem[]}
+   */
+  // eslint-disable-next-line no-unused-vars
+  getConfigProblems (extData) {
+    // shoud override this method if special validation is necessary for this extension type
+    return [];
+  }
+
+  /**
+   * @param {string} extName
+   * @param {ExtData<ExtType>} extData
+   * @param {ExtensionConfigMutationOpts} [opts]
+   * @returns {Promise<void>}
+   */
+  async addExtension (extName, extData, {write = true} = {}) {
+    await this.io.addExtension(this.extensionType, extName, extData);
+    if (write) {
+      await this.io.write();
+    }
+  }
+
+  /**
+   * @param {ExtName<ExtType>} extName
+   * @param {ExtData<ExtType>} extData
+   * @param {ExtensionConfigMutationOpts} [opts]
+   * @returns {Promise<void>}
+   */
+  async updateExtension (extName, extData, {write = true} = {}) {
+    this.installedExtensions[extName] = {
+      ...this.installedExtensions[extName],
+      ...extData,
+    };
+    if (write) {
+      await this.io.write();
+    }
+  }
+
+  /**
+   * @param {ExtName<ExtType>} extName
+   * @param {ExtensionConfigMutationOpts} [opts]
+   * @returns {Promise<void>}
+   */
+  async removeExtension (extName, {write = true} = {}) {
+    delete this.installedExtensions[extName];
+    if (write) {
+      await this.io.write();
+    }
+  }
+
+  /**
+   * @param {ExtName<ExtType>[]} [activeNames]
+   * @returns {void}
+   */
+  // eslint-disable-next-line no-unused-vars
+  print (activeNames) {
+    if (_.isEmpty(this.installedExtensions)) {
+      log.info(
+        `No ${this.configKey} have been installed in ${this.appiumHome}. Use the "appium ${this.extensionType}" ` +
+          'command to install the one(s) you want to use.',
+      );
+      return;
+    }
+
+    log.info(`Available ${this.configKey}:`);
+    for (const [
+      extName,
+      extData,
+    ] of /** @type {[string, ExtData<ExtType>][]} */ (
+        _.toPairs(this.installedExtensions)
+      )) {
+      log.info(`  - ${this.extensionDesc(extName, extData)}`);
+    }
+  }
+
+  /**
+   * Returns a string describing the extension. Subclasses must implement.
+   * @param {ExtName<ExtType>} extName - Extension name
+   * @param {ExtData<ExtType>} extData - Extension data
+   * @returns {string}
+   * @abstract
+   */
+  // eslint-disable-next-line no-unused-vars
+  extensionDesc (extName, extData) {
+    throw new Error('This must be implemented in a subclass');
+  }
+
+  /**
+   * @param {string} extName
+   * @returns {string}
+   */
+  getInstallPath (extName) {
+    return path.join(this.appiumHome, 'node_modules', this.installedExtensions[extName].pkgName);
+  }
+
+  /**
+   * Loads extension and returns its main class (constructor)
+   * @param {ExtName<ExtType>} extName
+   * @returns {ExtClass<ExtType>}
+   */
+  require (extName) {
```

Comment:
mayt this method shadow the original `require` global?

Reply:
Only if you dereference it, I suppose.  old code